xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/match/BlockNavigator.java

```diff
@@ -239,15 +239,15 @@ public BlockNavigator(BlockMatcher matcher)
                     nextBlock = block.getParent();
                     break;
                 // DESCENDANT
+                /* In order to reduce number of lines from 9 to 4,
+                   all variables with the exception of axes were removed from the code.
+                   nextBlock = block.getChildren().get(0)
+                   children = block.getChildren()
+                 */
                 case CHILD:
-                    List<Block> children = block.getChildren();
-                    if (!children.isEmpty()) {
-                        nextBlock = children.get(0);
+                    if (!block.getChildren().isEmpty()) {
                         axes = Axes.FOLLOWING_SIBLING;
-                        if (this.matcher.match(nextBlock)) {
-                            return (T) nextBlock;
-                        }
-                    }
+                        if (this.matcher.match(nextBlock)) { return (T) block.getChildren().get(0); } }
```

Comment:
Just removing the newlines is cheating :)

This does not follow XWiki code style so you are changing a violation for another here.